lib/Parser.js

```diff
@@ -37,6 +38,107 @@ const POSSIBLE_AST_OPTIONS = [{
 	}
 }];
 
+class StackedSetMap {
```

Comment:
This is pretty neat. Does it have any interfile scope dependencies? If not, separate file?